ddtrace/internal/writer.py

```diff
@@ -159,7 +159,7 @@ def stop(self, timeout=None):
         return
 
     def write(self, spans):
-        # type: (List[Span]) -> None
+        # type: (Optional[List[Span]]) -> None
```

Comment:
is this actually optional or can it just be empty?

Reply:
e.g. `write([])` vs `write(None)`, not idea why we should ever have the latter

Reply:
Technically I think it's possible to pass None which is why I wanted to be defensive (just in case other code down the line assume that spans is not None), but it also makes sense that we should never actually pass None into write - I'll change this to be non-optional

Reply:
it is possible, but yeah, I don't think we want that as our contract.

e.g. I think we should assume we can swap `if not spans:` out with `if not len(spans):` and not have any errors.

If there are places that are explicitly passing in `None`, then we probably need to update those calls

Reply:
Agreed that we don't want it but it's possible users are passing `None` to `tracer.write()` and it's a public API.

Reply:
> Agreed that we don't want it but it's possible users are passing `None` to `tracer.write()` and it's a public API.

Not sure I get what/if any suggestion from this? Should we keep as `Optional`?

Reply:
Yeah I'm in favour of typing what it permitted at the moment (`Optional`) here and then we can separately restrict the types and document if necessary.

Reply:
So just to confirm, do we want to keep the type hints as Optional? Are there any other steps that I can put into another JIRA ticket (i.e. document or restrict types)?